Source/WebCore/rendering/RenderLayer.h

```diff
@@ -557,29 +557,39 @@ class RenderLayer final : public CanMakeSingleThreadWeakPtr<RenderLayer>, public
     bool behavesAsFixed() const { return m_behavesAsFixed; }
 
     struct PaintedContentRequest {
-        void makeStatesUndetermined()
-        {
-            if (hasPaintedContent == RequestState::Unknown)
-                hasPaintedContent = RequestState::Undetermined;
-        }
-
         void setHasPaintedContent() { hasPaintedContent = RequestState::True; }
+        void makePaintedContentUndetermined() { hasPaintedContent = RequestState::Undetermined; }
+        bool probablyHasPaintedContent() const { return hasPaintedContent == RequestState::True || hasPaintedContent == RequestState::Undetermined; }
+        bool isPaintedContentSatisfied() const { return hasPaintedContent != RequestState::Unknown; }
 
-        bool needToDeterminePaintedContentState() const { return hasPaintedContent == RequestState::Unknown; }
+#if HAVE(HDR_SUPPORT)
+        void setHasPaintedHDRContent() { hasPaintedHDRContent = RequestState::True; }
+        void makePaintedHDRContentDontCare() { hasPaintedContent = RequestState::DontCare; }
+        bool probablyHasPaintedHDRContent() const { return hasPaintedHDRContent == RequestState::True || hasPaintedHDRContent == RequestState::Undetermined; }
+        bool isPaintedHDRContentSatisfied() const { return hasPaintedHDRContent != RequestState::Unknown; }
+#endif
 
-        bool probablyHasPaintedContent() const { return hasPaintedContent == RequestState::True || hasPaintedContent == RequestState::Undetermined; }
-        
-        bool isSatisfied() const { return hasPaintedContent != RequestState::Unknown; }
+        bool isSatisfied() const
```

Comment:
This still needs to have an "or".

I think the fact that we're struggling to get this right makes it clear that was need to be able to write tests for this.